### PR TITLE
Nfdiv 1177 generate sole divorce application 2

### PR DESCRIPTION
**Pull request by mapingo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1178

### Change description ###
Add sole divorce application generation

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java [hunk 1]
```diff
@@ -64,12 +64,62 @@ void shouldReturnEmptyListOfJurisdictionsIfNoJurisdictionConnectsSelected() {
...
+                + "resided there for at least six months immediately prior to the application"),
+            new Connection("the applicant and respondent are both domiciled in England and Wales"),
+            new Connection("the applicant and respondent registered as civil partners of each other in England "
+                + "or Wales or, in the case of a same sex couple, married each other under the law of England and Wales "
+                + "and it would be in the interests of justice for the court to assume jurisdiction in this case")
+        );
+    }
+
+    @Test
+    void shouldReturnEmptyListOfJurisdictionsForSoleIfNoJurisdictionConnectsSelected() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java:106]: I think if juridiction list is empty we should throw an exception as we cannot apply for divorce without jurisdiction selection. So ideally template provider should throw an error

File: src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java [hunk 2]
```diff
@@ -64,12 +64,62 @@ void shouldReturnEmptyListOfJurisdictionsIfNoJurisdictionConnectsSelected() {
...
+        final var application = Application.builder().build();
+        application.getJurisdiction().setConnections(emptySet());
+
+        final var result = applicationTemplateDataProvider.deriveSoleJurisdictionList(application);
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    void shouldReturnEmptyListOfJurisdictionsForSoleIfJurisdictionConnectionsIsNull() {
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java:117]: same here should throw an error I think

File: src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java (hunk 1 above)
Comment (mapingo) [reply src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java]: Added exception 
https://github.com/hmcts/nfdiv-case-api/pull/747/commits/408187f6680fd66c7b637bf28f8a54ac43c8572a

File: src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java (hunk 2 above)
Comment (mapingo) [reply src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java]: Added exception
https://github.com/hmcts/nfdiv-case-api/pull/747/commits/408187f6680fd66c7b637bf28f8a54ac43c8572a

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**